cluster/validation.go

```diff
@@ -731,3 +738,19 @@ func getClusterVersion(version string) (semver.Version, error) {
 	}
 	return parsedVersion, nil
 }
+
+// warnWeaveDeprecation prints a deprecation warning if version is higher than 1.27
+func warnWeaveDeprecation(k8sVersion string) error {
+	version, err := util.StrToSemVer(k8sVersion)
+	if err != nil {
+		return fmt.Errorf("failed to translate: %s to semver notation: %w", k8sVersion, err)
+	}
+	version127, err := util.StrToSemVer("v1.27.0")
+	if err != nil {
+		return fmt.Errorf("failed to tranlate v1.27.0 to semver notation: %w", err)
+	}
+	if !version.LessThan(*version127) {
```

Comment:
We scope this to 1.27 and up because we only support and expect users to first upgrade to new minor (so 1.25->1.26->1.27) to see this message? Is there a problem to warn every user of Weave that it will be removed in v1.30.x?

Reply:
No problem but this is was was agreed with the PM after a few conversations

Reply:
Can we include those agreements in the issue/PR? We can't review things if the agreements or decisions are not recorded which can be matched with the implementation. I am not here to override PM decisions so we can leave it for now.